judo-tatami-jsl-workflow-maven-plugin/src/main/java/hu/blackbelt/judo/tatami/jsl/workflow/maven/plugin/AbstractJslDslWorkflowProjectMojo.java

```diff
@@ -0,0 +1,284 @@
+
+    @Component
+    public RepositorySystem repoSystem;
+
+    @Parameter(defaultValue = "${repositorySystemSession}", readonly = true, required = true)
+    public RepositorySystemSession repoSession;
+
+    @Parameter(defaultValue = "${project.remoteProjectRepositories}", readonly = true, required = true)
+    public List<RemoteRepository> repositories;
+
+    @Parameter(defaultValue = "${plugin}", readonly = true)
+    private PluginDescriptor pluginDescriptor;
+
+    @Parameter(property = "actors")
+    private List<String> actors;
+
+    @Parameter(property = "scanSources", defaultValue = "false")
+    public Boolean scanSources = false;
+
+    @Parameter(property = "srcModelTarget", defaultValue = "${project.basedir}/target/classes/model")
+    public File srcModelTarget;
+
+    @Parameter(property = "sources", defaultValue = "${project.basedir}/src/main/model")
+    public List<String> sources;
+
+    @Parameter(property = "modelNames")
+    public List<String> modelNames;
+
+    @Parameter(property = "jsl")
+    public String jsl;
+
+    Set<URL> classPathUrls = new HashSet<>();
+
+    ArtifactResolver artifactResolver;
+
+
+    public abstract void performExecution(JslDslModel jslModel) throws MojoExecutionException, MojoFailureException;
+
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+
+        artifactResolver = ArtifactResolver.builder()
+                .log(getLog())
+                .project(project)
+                .repoSession(repoSession)
+                .repositories(repositories)
+                .repoSystem(repoSystem)
+                .build();
+
+        // Needed for to access project's dependencies.
+        // Info: http://blog.chalda.cz/2018/02/17/Maven-plugin-and-fight-with-classloading.html
+        try {
+            setContextClassLoader(artifactResolver);
+        } catch (Exception e) {
+            throw new MojoExecutionException("Failed to set classloader", e);
+        }
+        JslDslModel jslDslModel;
+
+        if (jsl == null || jsl.trim().isBlank()) {
+            List<File> jslFiles = resolveJslFiles(artifactResolver);
+            if (jslFiles.isEmpty()) {
+                getLog().warn("No JSL files presented to process");
+            }
+
+            XtextResourceSet resourceSet = JslParser.loadJslFromFile(jslFiles);
+            Collection<ModelDeclaration> allModelDeclcarations = JslParser.getAllModelDeclarationFromXtextResourceSet(resourceSet);
+
+            Collection<String> effectiveModelsNames = calculateEffectiveModelDeclarations(allModelDeclcarations);
+            Collection<ModelDeclaration> effectiveModels = allModelDeclcarations.stream().filter(m -> effectiveModelsNames.contains(m.getName())).collect(Collectors.toList());
+
+            if (effectiveModels.size() > 1) {
+                throw new MojoExecutionException("Multiple root model, please remove or define the correct ones in 'modelNames' -  "
```

Comment:
If having multiple root models is a problem why are we requesting the caller to define multiple model (name?)s?

Reply:
Because the models can import other models. The goal that one model can be defined as root